lib/ngx/balancer.lua

```diff
@@ -101,4 +104,23 @@ function _M.get_last_failure()
 end
 
 
+function _M.set_timeout(connect_timeout, send_timeout, read_timeout)
```

Comment:
You right. In our business case, it's adjust read timeout only. 
Because of change timeout will allocating new memory for per request, so I write a method who can set a series variable of timeout. 
I will fix it when we resolve the memory problem.